pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarServiceNameResolver.java

```diff
@@ -52,9 +52,8 @@ public InetSocketAddress resolveHost() {
         if (list.size() == 1) {
             return list.get(0);
         } else {
-            CURRENT_INDEX_UPDATER.getAndUpdate(this, last -> (last + 1) % list.size());
-            return list.get(currentIndex);
-
+            int originalIndex = CURRENT_INDEX_UPDATER.getAndUpdate(this, last -> (last + 1) % list.size());
+            return list.get((originalIndex + 1) % list.size());
```

Comment:
What was the purpose of this change? It seems that the index is incremented by 2 each time. That would skip some entries in the input. It seems that the previous logic had a bug and should have been `return list.get(CURRENT_INDEX_UPDATER.getAndUpdate(this, last -> (last + 1) % list.size()));` to prevent races.

Reply:
I'll check with a unit test